#!/usr/bin/env python
PACKAGE = "franka_custom_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness_X", double_t, 0, "Cartesian translational stiffness", 400, 0, 2000)
gen.add("translational_stiffness_Y", double_t, 0, "Cartesian translational stiffness", 400, 0, 2000)
gen.add("translational_stiffness_Z", double_t, 0, "Cartesian translational stiffness", 400, 0, 6000)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 30, 0, 100)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0, 0, 100)
gen.add("max_force_X", double_t, 0, "Cartesian force", 30, 0, 50)
gen.add("max_force_Y", double_t, 0, "Cartesian force", 30, 0, 50)
gen.add("max_force_Z", double_t, 0, "Cartesian force", 30, 0, 50)
gen.add("max_cartesian_torque", double_t, 0, "Cartesian torque", 30, 0, 50)


exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param_demonstration"))
